### PR TITLE
pep8 adjustments

### PR DESCRIPTION
**Pull request by bgruening**
xref: https://github.com/compomics/moFF/issues/34

@Maux82 do you have any test cases which you can run against this?

### CODE REVIEW

File: moff.py [hunk 1]
```diff
@@ -40,700 +39,749 @@
...
+                comps["O"] += 1
+        else:
+            # fixed and variable mod are both in the sequence
+            comps = Counter(
+                list(chain(*[list(std_aa_comp[aa].elements()) for aa in row.peptide])))
+            if ('<' in row.mod_peptide):
+                # check only if modificatio are present.
+                # Ffor the future use dthe tag_mod_sequence_delimiter use in moFF_setting
+                for ptm in ptm_map.keys():
+                    ptm_c = row.mod_peptide.count(ptm)
```
Comment (bgruening) [inline moff.py:782]: @Maux82 this does not seem to work?

File: moff.py [hunk 2]
```diff
@@ -40,700 +39,749 @@
...
+                temp['te'] = (data_ms2['rt'] / 60) + h_rt_w
+            temp.drop('rt', 1, inplace=True)
+            xic_data = get_xic_data(
+                flag_mzml, flag_windows, temp, loc_output, name_file, txic_path, loc, 0)
+            data_ms2.reset_index(inplace=True)
+            data_ms2[['10p_noise', '5p_noise', 'SNR', 'intensity', 'log_L_R', 'log_int', 'lwhm', 'rt_peak', 'rwhm']] = data_ms2.apply(
+                lambda x: compute_peak_simple(x, xic_data, log, mbr_flag, h_rt_w, s_w, s_w_match, offset_index, moff_pride_flag, -1, -1, 1), axis=1)
+
+    except Exception as e:
+        traceback.print_exc()
```
Comment (bgruening) [inline moff.py:648]: This should be cleanup-up I think.

File: moff.py [hunk 3]
```diff
@@ -40,700 +39,749 @@
...
+                data_ms2[['Erro_RelIntensity_TheoExp', 'RT_drift', 'delta_log_int', 'delta_rt', 'rankcorr']] = data_ms2.apply(lambda x: estimate_on_match_peak(
+                    x, all_isotope_df, estimate_flag, moff_pride_flag, log, rt_drift, err_ratio_int, xic_data, mbr_flag, h_rt_w, s_w, s_w_match, offset_index), axis=1)
+            if estimate_flag != 1:
+                data_ms2 = data_ms2[(data_ms2[['10p_noise', '5p_noise', 'SNR', 'intensity',
+                                               'log_L_R', 'log_int', 'lwhm', 'rt_peak', 'rwhm']] != -1).all(1)]
+        else:
+            # not match  filter
+            temp = data_ms2[['mz', 'rt']].copy()  # strange cases
+            temp.ix[:, 'tol'] = int(tol)
+            if moff_pride_flag == 1:
```
Comment (bgruening) [inline moff.py:634]: all these flags should be booleans - True/False I think.

File: moff.py [hunk 4]
```diff
@@ -40,700 +39,749 @@
...
+    if estimate_flag == 1:
+        # add extra filed if I am in a estimate mode
+        data_ms2["Erro_RelIntensity_TheoExp"] = -1
+        data_ms2["rankcorr"] = -1
+        data_ms2["RT_drift"] = -1
+        data_ms2["delta_rt"] = -1
+        data_ms2["delta_log_int"] = -1
+    # set mbr_flag
+    if 'matched' in data_ms2.columns:
+        mbr_flag = 1
```
Comment (bgruening) [inline moff.py:595]: boolean

File: moff.py [hunk 5]
```diff
@@ -40,700 +39,749 @@
...
+    log.critical('Estimated distribition ratio exp. int. left isotope vs. monoisotopic isotope %r ',
+                 ms2_data[ms2_data['delta_log_int'] != -1]['delta_log_int'].describe())
+    error_relInr = ms2_data[ms2_data['Erro_RelIntensity_TheoExp'] != -1]['Erro_RelIntensity_TheoExp'].quantile(quantile_value)
+    rt_drift = ms2_data[ms2_data['RT_drift'] != -1]['RT_drift'].quantile(quantile_value)
+    ratio_log_int = ms2_data[ms2_data['delta_log_int'] != -1]['delta_log_int'].quantile(quantile_value)
+    return (rt_drift, error_relInr, ratio_log_int)
+
+
+def compute_match_peak_quality_measure(input_data, moff_pride_flag, log):
+    mad_diff_int = np.mean(abs((input_data['intensity'] / input_data['intensity'].sum()) - (
```
Comment (bgruening) [inline moff.py:408]: It would speed up things if we just calculate the sum() once, save it and reuse it.

File: moff.py [hunk 6]
```diff
@@ -40,700 +39,749 @@
...
+    log.critical('Writing peptide_summary intensity file')
+    df.to_csv(os.path.join(loc_output, "peptide_summary_intensity_" +
+                           tag_filename + ".tab"), sep='\t', index=False)
+    return 1
 
 
+def save_moff_apex_result(list_df, result):
+    try:
+        xx = []
+        for df_index in range(0, len(list_df)):
```
Comment (bgruening) [inline moff.py:124]:
```
for result in list_df:
   if result.get()[1] == -1:
```

seems to be easier.

File: moff.py [hunk 7]
```diff
@@ -40,700 +39,749 @@
...
-	if type_mapping == 'col_must_have_apex':
-		data.rename(columns={'sequence': 'peptide', 'modified sequence': 'mod_peptide' ,'measured charge': 'charge', 'theoretical mass': 'mass',
-		                     'protein(s)': 'prot', 'm/z': 'mz'}, inplace=True)
-	return data, data.columns.values.tolist()
+def compute_peptide_matrix(loc_output, log, tag_filename):
+    name_col = []
+    name_col.append('prot')
+    d = []
+    if not glob.glob(loc_output + '/*_moff_result.txt'):
+        return -1
```
Comment (bgruening) [inline moff.py:72]: return False

File: moff.py [hunk 8]
```diff
@@ -40,700 +39,749 @@
...
+        df.ix[np.intersect1d(df.index, grouped.groups.keys()), 0] = grouped.agg({'prot': 'max', 'intensity': 'sum'})[
+            'prot']
+    # print df.head(5)
+    df.reset_index(level=0, inplace=True)
+    df = df.fillna(0)
+    df.rename(columns={'index': 'peptide'}, inplace=True)
+    log.critical('Writing peptide_summary intensity file')
+    df.to_csv(os.path.join(loc_output, "peptide_summary_intensity_" +
+                           tag_filename + ".tab"), sep='\t', index=False)
+    return 1
```
Comment (bgruening) [inline moff.py:118]: return True

File: moff.py [hunk 9]
```diff
@@ -40,700 +39,749 @@
...
+
+        return (rt_list, runid_list)
+    else:
+        # in case of raw file  I put to -1 -1 thm result
+        return (-1, -1)
+
+
+def mzML_get_all(temp, tol, loc, run, rt_list1, runid_list1):
+    app_list = []
+    for index_ms2, row in temp.iterrows():
```
Comment (bgruening) [inline moff.py:202]: temp is a non-ideal name here.
Can we also add a docstring to document what this function is doing.

File: moff.py [hunk 10]
```diff
@@ -40,700 +39,749 @@
...
+    timeDependentIntensities = []
+    minpos = bisect.bisect_left(rt_list, minRT)
+    maxpos = bisect.bisect_left(rt_list, maxRT)
+
+    for specpos in range(minpos, maxpos):
+        specid = runid_list[specpos]
+        spectrum = run[specid]
+        if spectrum['scan start time'] > maxRT:
+            break
+        if spectrum['scan start time'] > minRT and spectrum['scan start time'] < maxRT:
```
Comment (bgruening) [inline moff.py:224]: if we reach this point it is known that `and spectrum['scan start time'] < maxRT` is true, isn't it?

File: moff.py [hunk 11]
```diff
@@ -40,700 +39,749 @@
...
+            lower_index = bisect.bisect(
+                spectrum.peaks, (float(MZValue - ppmPrecision * MZValue), None))
+            upper_index = bisect.bisect(
+                spectrum.peaks, (float(MZValue + ppmPrecision * MZValue), None))
+            maxI = 0.0
+            for sp in spectrum.peaks[lower_index: upper_index]:
```
Comment (bgruening) [inline moff.py:230]: can we use max() here like `max(lis, key=lambda x:x[1])`

File: moff.py [hunk 12]
```diff
@@ -40,700 +39,749 @@
...
-		c = x.name
-	data_xic = xic_array[c]
-	if rt_match_peak > -1 :
-		time_w= rt_match_peak
-	else:
-		time_w= x['rt']
-        if moff_pride_flag == 0 :
+    if os.path.isfile(file_to_check):
+        os.remove(file_to_check)
+        return 1
```
Comment (bgruening) [inline moff.py:246]: True

File: moff.py [hunk 13]
```diff
@@ -40,700 +39,749 @@
...
+    else:
+        return -1
```
Comment (bgruening) [inline moff.py:248]: False

File: moff.py [hunk 14]
```diff
@@ -40,700 +39,749 @@
...
+        return 0
+
+
+def compute_log_LR(data_xic, index, v_max, disc):
+    log_time = [-1, -1]
+    c_left = 0
+    find_5 = False
+    stop = False
+    while c_left <= (index - 1) and not stop:
+        if not find_5 and (data_xic.ix[(index - 1) - c_left, 1] <= (disc * v_max)):
```
Comment (bgruening) [inline moff.py:265]: we can precalculate `data_xic.ix[(index - 1) - c_left` and reuse it.

File: moff.py [hunk 15]
```diff
@@ -40,700 +39,749 @@
...
-				'rwhm': time_point[1] ,
-				'5p_noise': pnoise_5,
-				'10p_noise':pnoise_10,
-				'SNR':SNR[0],
-				'log_L_R': log_L_R,
-				'log_int': np.log2(val_max)[0] })
+            # standar cases rt must be in second
+            time_w = time_w / 60
+    # print time_w, x['rt'] , moff_pride_flag, rt_match_peak,time_w, s_w,s_w_match
+    if mbr_flag == 0:
```
Comment (bgruening) [inline moff.py:297]: with boolean this becomes `if mbr_flag:`

File: moff.py [hunk 16]
```diff
@@ -40,700 +39,749 @@
...
+        else:
+            temp_w = s_w
+    if data_xic[(data_xic['rt'] > (time_w - temp_w)) & (data_xic['rt'] < (time_w + temp_w))].shape[0] >= 1:
+        # data_xic[(data_xic['rt'] > (time_w - temp_w)) & (data_xic['rt'] < (time_w + temp_w))].to_csv('thermo_testXIC_'+str(c)+'.txt',index=False,sep='\t')
+        ind_v = data_xic.index
+        pp = data_xic[data_xic["intensity"] == data_xic[(data_xic['rt'] > (
+            time_w - temp_w)) & (data_xic['rt'] < (time_w + temp_w))]['intensity'].max()].index
+        pos_p = ind_v[pp]
+        if pos_p.values.shape[0] > 1:
+            print 'error'
```
Comment (bgruening) [inline moff.py:314]: this should be a log.warning or something similar.

File: moff.py [hunk 17]
```diff
@@ -40,700 +39,749 @@
...
+            return pd.Series({'intensity': -1, 'rt_peak': -1, 'lwhm': -1, 'rwhm': -1, '5p_noise': -1, '10p_noise': -1, 'SNR': -1, 'log_L_R': -1, 'log_int': -1})
+        val_max = data_xic.ix[pos_p, 1].values
+        # if filt_flag ==1:
+        # log.info('peptide at line %i -->  MZ: %4.4f RT: %4.4f matched (yes=1/no=0): %i Peak detected  ',(offset_index + c +2), x['mz'], time_w,x['matched'])
+    else:
+        if filt_flag == 1:
+            if ('matched'in x.axes[0].tolist()):
```
Comment (bgruening) [inline moff.py:321]: brackets can be removed.

File: moff.py [hunk 18]
```diff
@@ -40,700 +39,749 @@
...
+                          'SNR': -1,
+                          'log_L_R': -1,
+                          'log_int': -1})
+    pnoise_5 = np.percentile(data_xic[(data_xic['rt'] > (
+        time_w - h_rt_w)) & (data_xic['rt'] < (time_w + h_rt_w))]['intensity'], 5)
+    pnoise_10 = np.percentile(data_xic[(data_xic['rt'] > (
+        time_w - h_rt_w)) & (data_xic['rt'] < (time_w + h_rt_w))]['intensity'], 10)
+    # find the lwhm and rwhm
+    time_point = compute_log_LR(data_xic, pos_p[0], val_max, 0.5)
+    if (time_point[0] * time_point[1] == 1) or (time_point[0] * time_point[1] < 0):
```
Comment (bgruening) [inline moff.py:345]: brackets can be removed

File: moff.py [hunk 19]
```diff
@@ -40,700 +39,749 @@
...
+        time_point = compute_log_LR(data_xic, pos_p[0], val_max, 0.7)
+
+    if time_point[0] == -1 or time_point[1] == -1:
+        # keep the shape measure to -1 in case on txo point are -1
+        log_L_R = -1
+    else:
+        log_L_R = np.log2(
+            abs(time_w - time_point[0]) / abs(time_w - time_point[1]))
+
+    if (pnoise_5 == 0 and pnoise_10 > 0):
```
Comment (bgruening) [inline moff.py:356]: here as well ...

File: moff.py [hunk 20]
```diff
@@ -40,700 +39,749 @@
...
+    set_logger(log_file)
+    myPool = multiprocessing.Pool(multiprocessing.cpu_count())
+    sample = df[df['matched'] == 0].sample(frac=sample_size)
+    log.critical(
+        'quality measures estimation  using %r  MS2 ident. peptides randomly sampled' % sample.shape[0])
+    data_split = np.array_split(sample, multiprocessing.cpu_count())
+    result = {}
+    offset = 0
+    # run matchinf filtering for
+    for df_index in range(0, len(data_split)):
```
Comment (bgruening) [inline moff.py:389]: for result in data_split:

File: moff.py (hunk 1 above)
Comment (Maux82) [reply moff.py:782]: Actually the support for mzML file has not been tested at all. There is a new version pymzML library that I d like to use but supoport only python 3. So first I have to make it python 3 compatible.

**Review by Maux82 — APPROVED** [review]
Ok

**Review by Maux82 — APPROVED** [review]
ok

**Review by Maux82 — APPROVED** [review]
ok

**Review by Maux82 — APPROVED** [review]
ok

**Review by Maux82 — APPROVED** [review]

**Review by Maux82 — APPROVED** [review]

File: moff.py (hunk 5 above)
Comment (Maux82) [reply moff.py:408]: yes :) Good idea.

File: moff.py (hunk 3 above)
Comment (Maux82) [reply moff.py:634]: fixed

File: moff.py (hunk 4 above)
Comment (Maux82) [reply moff.py:595]: Fixed.

File: moff.py (hunk 16 above)
Comment (Maux82) [reply moff.py:314]: fixed

File: moff.py (hunk 17 above)
Comment (Maux82) [reply moff.py:321]: ok

File: moff.py (hunk 19 above)
Comment (Maux82) [reply moff.py:356]: ok

File: moff.py (hunk 7 above)
Comment (Maux82) [reply moff.py:72]: ok

File: moff.py (hunk 8 above)
Comment (Maux82) [reply moff.py:118]: ok

File: moff.py (hunk 12 above)
Comment (Maux82) [reply moff.py:246]: ok

File: moff.py (hunk 15 above)
Comment (Maux82) [reply moff.py:297]: ok fixed

File: moff.py (hunk 3 above)
Comment (bgruening) [reply moff.py:634]: cool, thanks for working on it ...

File: moff.py (hunk 6 above)
Comment (Maux82) [reply moff.py:124]: I changed and streamlined with : 
	
	for df_index in result:
			if result[df_index].get()[1] == -1:
				exit('Raw file not retrieved: wrong path or upper/low case mismatch')
			else:
				xx.append(result[df_index].get()[0])


eliminating also one parameter from the function:

def save_moff_apex_result(result)

File: moff_mbr.py [hunk 21]
```diff
@@ -17,526 +17,565 @@
...
+        # log.info('Custom list of peptide used  provided by the user in %s', args.rt_feat_file)
+        shared_pep_list = pd.read_csv(args.rt_feat_file, sep='\t')
+        shared_pep_list['mass'] = shared_pep_list['mass'].map('{:.4f}'.format)
+        shared_pep_list['code'] = shared_pep_list['peptide'].astype(
+            str) + '_' + shared_pep_list['mass'].astype(str)
+        list_shared_pep = shared_pep_list['code']
+        log.info('Custom list of peptide contains  %i ',
+                 list_shared_pep.shape[0])
+
+    for jj in aa:
```
Comment (bgruening) [inline moff_mbr.py:346]: @Maux82 I also found this here. You are iterating over aa here and then over out ... a few lines below you are iterating over the same aa. This has a complexity of n^4. I bet you can optimize this heavily to just use one for loop over aa, at least reducing it to n^2.

**Pull request merged**